proto/spire/api/agent/delegatedidentity/v1/delegatedidentity.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+package spire.api.agent.delegatedidentity.v1;
+option go_package = "github.com/spiffe/spire-api-sdk/proto/spire/api/agent/delegatedidentity/v1;delegatedidentityv1";
+
+import "spire/api/types/selector.proto";
+import "spire/api/types/x509svid.proto";
+
+// The delegatedIdentity service provides an interface to get the SVIDs of other
+// workloads on the host. This service is intended for use cases where a process
+// (different than the workload one) should access the workload's SVID to
+// perform actions on behalf of the workload. One example of is using a single
+// node instance of Envoy that upgrades TCP connections for different processes
+// running in such a node.
+//
+// The caller must be local and its identity must be listed in the allowed
+// clients on the spire-agent configuration.
+service DelegatedIdentity {
+    // Subscribe to get X.509-SVIDs for identities that match the given selectors.
+    // The lifetime of the subscription aligns to the lifetime of the stream.
+    rpc SubscribeToX509SVIDs(SubscribeToX509SVIDsRequest) returns (stream SubscribeToX509SVIDsResponse);
+
+    // Subscribe to get local and federated bundle.
+    // The lifetime of the subscription aligns to the lifetime of the stream.
+    rpc SubscribeToX509Bundle(SubscribeToX509BundleRequest) returns (stream SubscribeToX509BundleResponse);
+}
+
+// X.509 SPIFFE Verifiable Identity Document with the private key.
+message X509SVIDWithKey {
+    // The agent X509-SVID.
+    spire.api.types.X509SVID x509_svid = 1;
+
+    // Key encoding (DER PKCS#8).
+    bytes x509_svid_key = 2;
+
+    // Names of the trust domains that this workload should federates with.
+    repeated string federates_with = 3;
```

Comment:
In the workload api, we return this at the top level (i.e. as part of `SubscribeToX509SVIDsResponse`). The thinking was that we didn't want to slide into authorization land by telling workloads which trust domains a particular SVID should be used in conjunction with.

It's hard for me to tell whether that's the right thing to do here or not. There is also the question of consistency with the workload api, but we've already departed from that a bit.

Reply:
Hey @evan2645, I took a step back to follow the discuss in old https://github.com/spiffe/spire-api-sdk/pull/7. If I misunderstanding something, pls let me know. 

Are you saying that, today a workload can receive all trust domains (in `federates_with`) which is able to federate with, and this is not a good way to slide into authorization land?
Let's say that an alternative way would be a workload asking if its could federate with a specific trust domain? And, in spire, a check, answering true/false for the trust domain informed?

I don't know if I catch the ideia. Do you think that a critical point to an initial version?

Reply:
It is not a critical point. Since this API is not part of SPIFFE spec we can deprecate and change as needed.

Workload-to-identity is 1:N. The question here is whether federated_bundle should be in the workload or the identity.

I'm fine with this as-is, I just wanted to point it out in case someone had an opinion on it :)

Reply:
I think I'd slightly prefer it top-level in the response instead of duplicated for each identity, for both consistency sake with the Workload API and for reasons even has mentioned (sliding into authorization land).

Reply:
@azdagron In the case we put `federates_with` at top-level in the response (as the message `X509SVIDResponse`), we need to aggregate all the foreign trust domains (based on each registration entry) before sending them to workload, right?

Reply:
Yep.

Reply:
@azdagron @evan2645 maybe put the `federates_with` at top-level in the response make more sense once the Workload API do it in this way. At first I didn't get the ideia, but now I understand looking at `X509SVIDResponse`. By now I can't see a problem doing in this way :)
Tks again for all the discuss guys :D